docs/book/src/developer/conversion.md

```diff
@@ -0,0 +1,14 @@
+# Guide for API conversions
+
+## Introduction
+- [What are hubs and spokes?](https://book.kubebuilder.io/multiversion-tutorial/conversion-concepts.html)
+- [Reference Tutorial](https://book.kubebuilder.io/multiversion-tutorial/tutorial.html)
+
+## Conversion flow
+1. In each “spoke” version packages, we added the `+k8s:conversion-gen` directive pointing to the “hub” version package. It must be in `doc.go`. [Refer](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/main/api/v1alpha4/doc.go)
+2. In “hub” version package, we added `doc.go` file. [Refer](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/main/api/v1beta1/doc.go)
+3. We also added a var `localSchemeBuilder = &SchemeBuilder.SchemeBuilder` in `groupversion_info.go` so the auto-generated code would compile. [Refer](	https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/e55357a520089abc6ac2a9ca91a5b9cf0606bbbc/api/v1alpha4/groupversion_info.go#L37)
+4. Update the `Makefile` to run `conversion-gen` on the spoke versions to produce a `zz_generated.conversion.go` in the spoke version package. [Refer](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/main/Makefile#L105-L112)
```

Comment:
this is not required, we can just have `make generate-go-conversions-core` instruction